### PR TITLE
#1103 Fixed small issues with N-BEATS and N-HITS 

### PR DESCRIPTION
**Pull request by rijkvandermeulen**
Fixes an issue where num_stacks is used instead of self.num_stacks in the NBEATSModel. Also addresses a few mistakes in API reference docs.

### Summary

- Corrected mistake in NBEATS model where num_stacks was used iso self.num_stacks
- In the API reference documentation for both N-BEATS and N-HITS it was stated that num_layers is only used if generic _architecture is set to True. This is not the case, it's used all the times for N-BEATS and I believe N-HITS does not have a generic_architecture parameter. This has now been changed.

### Other Information
Fixed an unrelated minor typo somewhere else in the docs

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ Darts is still in an early development phase and we cannot always guarantee back
 - An issue with arguments being reverted for the `metric` function of gridsearch and backtest [#989](https://github.com/unit8co/darts/pull/989) by [Clara Grotehans](https://github.com/ClaraGrthns).
 - An error checking whether `fit()` has been called in global models [#944](https://github.com/unit8co/darts/pull/944) by [Julien Herzen](https://github.com/hrzn).
 - An error in Gaussian Process filter happening with newer versions of sklearn [#963](https://github.com/unit8co/darts/pull/963) by [Julien Herzen](https://github.com/hrzn).
+- An issue where num_stacks is used instead of self.num_stacks in the NBEATSModel. Also, a few mistakes in API reference docs [#1103](https://github.com/unit8co/darts/pull/1103) by [Rijk van der Meulen](https://github.com/rijkvandermeulen)
```
Comment (hrzn) [inline CHANGELOG.md:37]: +1 for adding an entry, however you should add it under the `Unreleased` section (not the already-release 0.20.0 version).

File: darts/models/forecasting/nbeats.py [hunk 2]
```diff
@@ -760,7 +759,7 @@ def __init__(
             self.num_stacks = 2
 
         if isinstance(layer_widths, int):
-            self.layer_widths = [layer_widths] * num_stacks
+            self.layer_widths = [layer_widths] * self.num_stacks
```
Comment (hrzn) [inline darts/models/forecasting/nbeats.py:762]: Could you also bring the same fix to `nhits.py` ?

File: darts/models/forecasting/nbeats.py [hunk 3]
```diff
@@ -574,7 +574,6 @@ def __init__(
             The number of blocks making up every stack.
         num_layers
             The number of fully connected layers preceding the final forking layers in each block of every stack.
-            Only used if `generic_architecture` is set to `True`.
```
Comment (hrzn) [inline darts/models/forecasting/nbeats.py:577]: Could you also change the docstring of `_NBEATSModule` ?

**Review by hrzn — COMMENTED** [review]
Looks good, only a couple of small things to fix

File: CHANGELOG.md (hunk 1 above)
Comment (rijkvandermeulen) [reply CHANGELOG.md]: Oh my bad, of course. I've changed it.

File: darts/models/forecasting/nbeats.py (hunk 2 above)
Comment (rijkvandermeulen) [reply darts/models/forecasting/nbeats.py:761]: Done

File: darts/models/forecasting/nbeats.py (hunk 3 above)
Comment (rijkvandermeulen) [reply darts/models/forecasting/nbeats.py:577]: Done

**Review by hrzn — APPROVED** [review]
Thanks!

**Pull request merged**